### PR TITLE
docs(fix): clarify how worker.version and minWorkerVersion comparison works

### PR DESCRIPTION
**Pull request by 317brian**
Add explanation that the value is treated as a string, so it works unintuitively if you don't provide the version in ISO format.

This PR has:
- [x] been self-reviewed.

### CODE REVIEW

**Review by zachjsh — APPROVED** [review]
LGTM

File: docs/configuration/index.md [hunk 1]
```diff
@@ -1078,7 +1078,7 @@ The following configs only apply if the Overlord is running in remote mode. For
 |Property|Description|Default|
 |--------|-----------|-------|
 |`druid.indexer.runner.taskAssignmentTimeout`|How long to wait after a task as been assigned to a MiddleManager before throwing an error.|PT5M|
-|`druid.indexer.runner.minWorkerVersion`|The minimum MiddleManager version to send tasks to. |"0"|
+|`druid.indexer.runner.minWorkerVersion`|The minimum MiddleManager version to send tasks to. Note that the version number is a string. This affects the expected behavior during certain operations like when it's compared against `druid.worker.version`. Specifically, the version comparison follows dictionary order . Use ISO8601 date format for the version so that later versions are recognized as being more recent. |"0"|
```
Comment (techdocsmith) [inline docs/configuration/index.md:1081]:
```suggestion
|`druid.indexer.runner.minWorkerVersion`|The minimum MiddleManager version to send tasks to. The version number is a string. This affects the expected behavior during certain operations like comparison against `druid.worker.version`. Specifically, the version comparison follows dictionary order. Use ISO8601 date format for the version to accommodate date comparisons. |"0"|
```

File: docs/configuration/index.md [hunk 2]
```diff
@@ -1369,7 +1369,7 @@ Middle managers pass their configurations down to their child peons. The MiddleM
 |`druid.indexer.runner.endPort`|Ending port used for peon processes, should be greater than or equal to `druid.indexer.runner.startPort` and less than 65536.|65535|
 |`druid.indexer.runner.ports`|A JSON array of integers to specify ports that used for peon processes. If provided and non-empty, ports for peon processes will be chosen from these ports. And `druid.indexer.runner.startPort/druid.indexer.runner.endPort` will be completely ignored.|`[]`|
 |`druid.worker.ip`|The IP of the worker.|localhost|
-|`druid.worker.version`|Version identifier for the MiddleManager.|0|
+|`druid.worker.version`|Version identifier for the MiddleManager. Note that the version number is a string. This affects the expected behavior during certain operations like when it's compared against `druid.indexer.runner.minWorkerVersion`. Specifically, the version comparison follows dictionary order. Use ISO8601 date format for the version so that later versions are recognized as being more recent.|0|
```
Comment (techdocsmith) [inline docs/configuration/index.md:1372]: see line 1081

**Review by techdocsmith — CHANGES_REQUESTED** [review]
small changes for style

File: docs/configuration/index.md (hunk 2 above)
Comment (317brian) [inline docs/configuration/index.md:1372]:
```suggestion
|`druid.worker.version`|Version identifier for the MiddleManager. The version number is a string. This affects the expected behavior during certain operations like comparison against `druid.indexer.runner.minWorkerVersion`. Specifically, the version comparison follows dictionary order. Use ISO8601 date format for the version to accomodate data comparisons.|0|
```

**Review by techdocsmith — APPROVED** [review]
LGTM

File: docs/configuration/index.md [hunk 3]
```diff
@@ -1369,7 +1369,7 @@ Middle managers pass their configurations down to their child peons. The MiddleM
 |`druid.indexer.runner.endPort`|Ending port used for peon processes, should be greater than or equal to `druid.indexer.runner.startPort` and less than 65536.|65535|
 |`druid.indexer.runner.ports`|A JSON array of integers to specify ports that used for peon processes. If provided and non-empty, ports for peon processes will be chosen from these ports. And `druid.indexer.runner.startPort/druid.indexer.runner.endPort` will be completely ignored.|`[]`|
 |`druid.worker.ip`|The IP of the worker.|localhost|
-|`druid.worker.version`|Version identifier for the MiddleManager.|0|
+|`druid.worker.version`|Version identifier for the MiddleManager. The version number is a string. This affects the expected behavior during certain operations like comparison against `druid.indexer.runner.minWorkerVersion`. Specifically, the version comparison follows dictionary order. Use ISO8601 date format for the version to accomodate date comparisons.|0|
```
Comment (techdocsmith) [inline docs/configuration/index.md:1372]:
```suggestion
|`druid.worker.version`|Version identifier for the MiddleManager. The version number is a string. This affects the expected behavior during certain operations like comparison against `druid.indexer.runner.minWorkerVersion`. Specifically, the version comparison follows dictionary order. Use ISO8601 date format for the version to accommodate date comparisons.|0|
```

**Pull request merged**